src/processor/Package.ts

```diff
@@ -330,4 +331,63 @@ export class Package {
       pullAt(sd.rules, rulesToRemove);
     });
   }
+
+  private removeDateRules(): void {
+    let allDatesMatch = true;
+    let date: string;
+    [...this.profiles, ...this.extensions, ...this.valueSets, ...this.codeSystems].forEach(sd => {
+      for (const rule of sd.rules) {
+        if (!allDatesMatch) break; // If any date hasn't matched the others, stop looking at other rules
+        if (
+          rule instanceof ExportableCaretValueRule &&
+          rule.caretPath === 'date' &&
+          rule.path === ''
+        ) {
+          const dateValue = rule.value as string; // If the rule assigns a date, the value will be a string.
+          if (!date) date = dateValue; // Set the date to match
+          if (date !== dateValue) {
+            allDatesMatch = false;
+          }
+        }
+      }
+    });
+
+    // If all dates are the same, are defined to the second, and are in GMT,
+    // we can assume they were set by the IG Publisher and can be safely removed.
+    const dateTimeRegex = /^([0-9]([0-9]([0-9][1-9]|[1-9]0)|[1-9]00)|[1-9]000)(-(0[1-9]|1[0-2])(-(0[1-9]|[1-2][0-9]|3[0-1])(T([01][0-9]|2[0-3]):[0-5][0-9]:([0-5][0-9]|60)(\.[0-9]+)?(Z|(\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00)))?)?)?$/;
+    const dateTimeMatch = date?.match(dateTimeRegex);
+    const usesGMT = dateTimeMatch?.[12] === '+00:00'; // dateTimeMatch[12] will hold timezone info if present
+    if (allDatesMatch && usesGMT) {
+      const DEFAULT_DATE = new ExportableCaretValueRule('');
+      DEFAULT_DATE.caretPath = 'date';
+      DEFAULT_DATE.value = date;
+      [...this.profiles, ...this.extensions].forEach(sd => {
+        const rulesToRemove: number[] = [];
+        sd.rules.forEach((rule, i) => {
+          if (isEqual(rule, DEFAULT_DATE)) {
+            rulesToRemove.push(i);
+          }
+        });
+        pullAt(sd.rules, rulesToRemove);
```

Comment:
I know this is a pattern we use in a lot of the optimizers -- and I think there is a reason for it in the others (?) -- but here, wouldn't it be valid just to use the much simpler:
```ts
sd.rules = sd.rules.filter(rule => !isEqual(rule, DEFAULT_DATE));
```

Reply:
And same for the two applications below.